compiler/resolution/functionResolution.cpp

```diff
@@ -5356,6 +5363,12 @@ static bool moveSupportsUnresolvedFunctionReturn(CallExpr* call) {
   return retval;
 }
 
+static bool isIfExprResult(CallExpr* call) {
```

Comment:
I think the LHS is "the if expr result" but call is something producing the if-expr.
So, I'm asking you to rename this function, to perhaps "moveToIfExprResult" or "setsIfExprResult" or something along those lines. Additionally, I think it should check that call is PRIM_MOVE or PRIM_ASSIGN.

Alternatively, you could keep the current name and just have it take in LHS of the move, since it's only called in resolveMove.